src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -163,20 +164,28 @@ static string ToGraphicShaderLevelDescription(int shaderLevel)
                 };
         }
 
-        private static void PopulateUnity(Protocol.Unity unity)
+        private void PopulateUnity(Protocol.Unity unity)
         {
-            unity.InstallMode = Application.installMode.ToString();
+            unity.InstallMode = _application.InstallMode.ToString();
         }
 
         private void PopulateTags(SentryEvent @event)
         {
-            @event.SetTag("unity.gpu.supports_instancing", SystemInfo.supportsInstancing ? "true" : "false");
-            @event.SetTag("unity.device.device_type", SystemInfo.deviceType.ToString());
-            @event.SetTag("unity.install_mode", Application.installMode.ToString());
+            @event.SetTag("unity.install_mode", _application.InstallMode.ToString());
 
-            if (_sentryOptions.SendDefaultPii)
+            if (_mainThreadData.SupportsDrawCallInstancing.HasValue)
             {
-                @event.SetTag("unity.device.unique_identifier", SystemInfo.deviceUniqueIdentifier);
+                @event.SetTag("unity.gpu.supports_instancing", _mainThreadData.SupportsDrawCallInstancing.Value ? "true" : "false");
+            }
+
+            if (_mainThreadData.DeviceType is not null)
+            {
+                @event.SetTag("unity.device.device_type", _mainThreadData.DeviceType);
+            }
+
+            if (_sentryOptions.SendDefaultPii && _mainThreadData.DeviceUniqueIdentifier is not null)
```

Comment:
What about this to avoid the double read (which is `extern` so P/Invoke):

```suggestion
            if (_sentryOptions.SendDefaultPii && _mainThreadData.DeviceUniqueIdentifier is {} deviceUniqueId)
```

Reply:
It's not extern anymore as it's cached in `MainThreadData` already.